### PR TITLE
refactor(experimental): extract common spec logic into new @solana/rpc-spec-types package

### PR DESCRIPTION
**Pull request by lorisleiva**
This PR extracts common types and helpers from `rpc-spec` and `rpc-subscriptions-spec` into a new `rpc-spec-types` package.

RPC packages reshuffle@2x (3)

### CODE REVIEW

**Comment by lorisleiva** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2146?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* **#2146** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2146?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2145** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2145?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2144** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2144?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2146?utm_source=stack-comment).

**Review by steveluscher — APPROVED** [review]
Added via Giphy

File: packages/rpc-subscriptions-spec/src/__tests__/rpc-subscription-test.ts [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
 import { RpcSubscriptionsApi } from '../rpc-subscriptions-api';
 import { RpcSubscriptionsRequest } from '../rpc-subscriptions-request';
 import { RpcSubscriptionsTransport } from '../rpc-subscriptions-transport';
 
-jest.mock('../rpc-message');
+// Partially mock the rpc-spec-types package.
+jest.mock('@solana/rpc-spec-types', () => ({
+    ...jest.requireActual('@solana/rpc-spec-types'),
+    createRpcMessage: jest.fn(),
+}));
```
Comment (buffalojoec) [inline packages/rpc-subscriptions-spec/src/__tests__/rpc-subscription-test.ts:12]: TIL you can do this!

**Review by buffalojoec — APPROVED** [review]
types

**Comment by lorisleiva** [comment]
✅ This PR was merged as part of https://github.com/solana-labs/solana-web3.js/pull/2168.

**Pull request closed without merging**